### PR TITLE
Bug fix in memory measurement

### PR DESCRIPTION
**Pull request by hbdeshmukh**
- Some NULL checks added without which we may have segmentation faults.

### CODE REVIEW

File: storage/AggregationOperationState.cpp [hunk 1]
```diff
@@ -949,9 +949,15 @@ void AggregationOperationState::finalizeHashTableImplThreadPrivate(
 std::size_t AggregationOperationState::getMemoryConsumptionBytes() const {
   std::size_t memory = getMemoryConsumptionBytesHelper(distinctify_hashtables_);
   memory += getMemoryConsumptionBytesHelper(group_by_hashtables_);
-  memory += collision_free_hashtable_->getMemoryConsumptionBytes();
-  memory += group_by_hashtable_pool_->getMemoryConsumptionPoolBytes();
-  memory += partitioned_group_by_hashtable_pool_->getMemoryConsumptionPoolBytes();
+  if (collision_free_hashtable_ != nullptr) {
```
Comment (cramja) [inline storage/AggregationOperationState.cpp]: Are these mutually exclusive? If so, could we use `else if` ?

Comment (hbdeshmukh) [inline storage/AggregationOperationState.cpp]: I think it is alright to have different if statements here.

**Comment by jianqiao** [comment]
LGTM! Merging.

**Comment by cramja** [comment]
+1

**Comment by jianqiao** [comment]
Have merged the branch but not reflected in github.

**Pull request merged**